src/ResourceManagement/Sql/SqlVirtualNetworkRuleOperationsImpl.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            try
+            {
+                var vnetInner = await this.sqlServerManager.Inner.VirtualNetworkRules
+                    .GetAsync(resourceGroupName, sqlServerName, name, cancellationToken);
+                return vnetInner != null ? new SqlVirtualNetworkRuleImpl(resourceGroupName, sqlServerName, vnetInner.Name, vnetInner, sqlServerManager) : null;
+            }
+            catch (CloudException ex) when (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+            catch (AggregateException ex) when ((ex.InnerExceptions[0] as CloudException).Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+        }
+
+        ///GENMHASH:48BA5938680939720DEA90858B8FC7E4:7B0BE68814CDBEBD16BFB8FD8569FD09
+        public async Task<Microsoft.Azure.Management.Sql.Fluent.ISqlVirtualNetworkRule> GetBySqlServerAsync(ISqlServer sqlServer, string name, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            try
+            {
+                var vnetInner = await sqlServer.Manager.Inner.VirtualNetworkRules
+                    .GetAsync(sqlServer.ResourceGroupName, sqlServer.Name, name, cancellationToken);
+                return vnetInner != null ? new SqlVirtualNetworkRuleImpl(vnetInner.Name, (SqlServerImpl)sqlServer, vnetInner, sqlServerManager) : null;
+            }
+            catch (CloudException ex) when (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+            catch (AggregateException ex) when ((ex.InnerExceptions[0] as CloudException).Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+        }
+
+        ///GENMHASH:4ABAEC77990815B01AA39D981ECF5CA5:5BC2CE295A48AC2AB6D73BB441883E14
+        public void DeleteBySqlServer(string resourceGroupName, string sqlServerName, string name)
+        {
+            Extensions.Synchronize(() => this.DeleteBySqlServerAsync(resourceGroupName, sqlServerName, name));
+        }
+
+        ///GENMHASH:8ACFB0E23F5F24AD384313679B65F404:97EADFA6DB9288740DDD782E3253A18B
+        public SqlVirtualNetworkRuleImpl Define(string name)
+        {
+            SqlVirtualNetworkRuleImpl result = new SqlVirtualNetworkRuleImpl(name, new Models.VirtualNetworkRuleInner(), this.sqlServerManager);
+            return (this.sqlServer != null) ? result.WithExistingSqlServer(this.sqlServer) : result;
+        }
+
+        public ISqlVirtualNetworkRule GetById(string id)
+        {
+            return Extensions.Synchronize(() => this.GetByIdAsync(id));
+        }
+
+        public Task<ISqlVirtualNetworkRule> GetByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
```

Comment:
you have bunch of NotImplemented methods here. Please revise.

Reply:
fixed